### ISSUE TITLE
reduce the first maven build time

### ISSUE THREAD

**Issue by vlukashov** [vaadin/base-starter-connect#46]
When building the starter project with empty maven caches it takes over 9 minutes on a high-end laptop with a 200Mbps connection. For comparison, for a JHipster-generated Spring Boot project with sufficiently more dependencies, it takes about 1.5 minutes.

This task is to investigate the slow build and propose corrective actions. The clean build of the Connect base Starter should be quicker than the clean build of a JHipster-generated project with simple default settings:

```
Welcome to JHipster v5.7.2

? Which *type* of application would you like to create? Monolithic application (recommended for simple projects)
? What is the base name of your application? jhipster
? What is your default Java package name? com.mycompany.myapp
? Do you want to use the JHipster Registry to configure, monitor and scale your application? No
? Which *type* of authentication would you like to use? JWT authentication (stateless, with a token)
? Which *type* of database would you like to use? SQL (H2, MySQL, MariaDB, PostgreSQL, Oracle, MSSQL)
? Which *production* database would you like to use? MySQL
? Which *development* database would you like to use? H2 with in-memory persistence
? Do you want to use the Spring cache abstraction? Yes, with the Ehcache implementation (local cache, for a single node)
? Do you want to use Hibernate 2nd level cache? Yes
? Would you like to use Maven or Gradle for building the backend? Maven
? Which other technologies would you like to use?
? Which *Framework* would you like to use for the client? React
? Would you like to enable *SASS* stylesheet preprocessor? No
? Would you like to enable internationalization support? No
? Besides JUnit and Jest, which testing frameworks would you like to use?
? Would you like to install other generators from the JHipster Marketplace? No
```

DoD:
 - there is a comment to this ticket with the research results (explaining the slow build)
 - there is a list of proposed corrective actions to make the build quicker than the target

**Comment by vlukashov** [comment]
Some ideas to check:
 - the Connect starter uses several Maven repos (including the `vaadin-prereleases`), that could be slower than using only the Maven central repo
 - is the plugins dependency tree required to build the Connect starter larger than that of the JHipster-generated project?

**Comment by Legioth** [comment]
One good way of finding out what is actually downloaded is to remove (or temporarily rename) your `~/.m2/` directory, running whatever you need to run and then inspect the contents of your new `~/.m2/` directory.

**Comment by vlukashov** [comment]
timebox for the research: 0.5 day

**Comment by qtdzz** [comment]
> Reason for the slow build

The order of repositories in pom.xml matters: we have `vaadin-prereleases` and `vaadin-snapshots` in pom file, so for each dependency, maven finds from those repositories first before falling back to maven central. I proposed a fix in #52 which reduces the first build time (with clean `.m2`) from **~15m** to **~1m40s** in my local machine.

>is the plugins dependency tree required to build the Connect starter larger than that of the JHipster-generated project?

No, our dependencies tree is smaller than the jhipster generated project.
